### PR TITLE
feat(auth): add device metadata to signed in data and credential store

### PR DESCRIPTION
**Pull request by div5yesh**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by tylerjroach — APPROVED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SRPCognitoActions.kt [hunk 1]
```diff
@@ -122,38 +114,18 @@ object SRPCognitoActions : SRPActions {
...
                 }
-                response?.let { respondToAuthChallengeResponse ->
-                    respondToAuthChallengeResponse.authenticationResult?.newDeviceMetadata?.let { deviceMetaData ->
-                        val confirmDeviceParams = DeviceMetaData(
-                            idToken = respondToAuthChallengeResponse.authenticationResult?.idToken
-                                ?: throw AuthException("Sign in failed", AuthException.TODO_RECOVERY_SUGGESTION),
-                            refreshToken = respondToAuthChallengeResponse.authenticationResult?.refreshToken
-                                ?: throw AuthException("Sign in failed", AuthException.TODO_RECOVERY_SUGGESTION),
-                            accessToken = respondToAuthChallengeResponse.authenticationResult?.accessToken
-                                ?: throw AuthException("Sign in failed", AuthException.TODO_RECOVERY_SUGGESTION),
```
Comment (gpanshu) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SRPCognitoActions.kt:133]: This change can not be intentional?

**Review by gpanshu — CHANGES_REQUESTED** [review]
Check SignInCognitoActions as it appears there was an intentional deletion there of when newdevicemeta is not null to make the confirmDevice call.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SRPCognitoActions.kt (hunk 1 above)
Comment (div5yesh) [reply aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SRPCognitoActions.kt:133]: It is. This handling is already done in SignInChallengeHelper. This was duplicate.

**Comment by div5yesh** [comment]
> Check SignInCognitoActions as it appears there was an intentional deletion there of when newdevicemeta is not null to make the confirmDevice call.

SigninChallengeHelper handles whether to send confirm device event or not: https://github.com/aws-amplify/amplify-android/pull/1971/files#diff-5637b40d344802c38b368feab2b624b02757ca79a9e595d7a3b27d06f904dd90R61

**Review by gpanshu — APPROVED** [review]

**Pull request merged**